instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/datasource/OpenTelemetryDataSource.java

```diff
@@ -40,29 +43,44 @@
 public class OpenTelemetryDataSource implements DataSource, AutoCloseable {
 
   private final DataSource delegate;
+  private final OpenTelemetry openTelemetry;
+  private final Instrumenter<DataSource, Void> instrumenter;
+
+  /**
+   * Create a OpenTelemetry DataSource wrapping another DataSource.
+   *
+   * @param delegate the DataSource to wrap
+   */
+  @Deprecated
+  public OpenTelemetryDataSource(DataSource delegate) {
+    this(delegate, GlobalOpenTelemetry.get());
+  }
 
   /**
    * Create a OpenTelemetry DataSource wrapping another DataSource. This constructor is primarily
    * used by dependency injection frameworks.
    *
    * @param delegate the DataSource to wrap
+   * @param openTelemetry the OpenTelemetry instance to setup for
    */
-  public OpenTelemetryDataSource(DataSource delegate) {
+  public OpenTelemetryDataSource(DataSource delegate, OpenTelemetry openTelemetry) {
     this.delegate = delegate;
+    this.openTelemetry = openTelemetry;
+    this.instrumenter = createInstrumenter(this.openTelemetry);
   }
 
   @Override
   public Connection getConnection() throws SQLException {
     Connection connection = wrapCall(delegate::getConnection);
     DbInfo dbInfo = computeDbInfo(connection);
-    return new OpenTelemetryConnection(connection, dbInfo);
+    return new OpenTelemetryConnection(connection, dbInfo, openTelemetry);
   }
 
   @Override
   public Connection getConnection(String username, String password) throws SQLException {
     Connection connection = wrapCall(() -> delegate.getConnection(username, password));
     DbInfo dbInfo = computeDbInfo(connection);
-    return new OpenTelemetryConnection(connection, dbInfo);
+    return new OpenTelemetryConnection(connection, dbInfo, openTelemetry);
```

Comment:
Each statement has to create its own instrumenter, right? Consider adding a `statementInstrumenter` field to this class, building it just once (instrumenter is stateless and can be reused) and passing it to the `OpenTelemetryConnection` constructor instead of `openTelemetry`.

Reply:
Got it!